tests/acceptance/features/apiLocks/lockFiles.feature

```diff
@@ -386,3 +409,93 @@ Feature: lock files
       | new              | exclusive  |
       | spaces           | shared     |
       | spaces           | exclusive  |
+
+
+  Scenario Outline: lock a file inside a folder shared by a link as anonymous user with edit permission
+    Given using <dav-path-version> DAV path
+    And user "Alice" has created folder "PARENT"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "PARENT/textfile0.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | PARENT   |
+      | permissions | change   |
+      | password    | %public% |
+    When the public locks "textfile0.txt" in the last public link shared folder using the new public WebDAV API setting the following properties
+      | lockscope | <lock-scope> |
+    Then the HTTP status code should be "200"
+    And 1 locks should be reported for file "PARENT/textfile0.txt" of user "Alice" by the WebDAV API
+    And user "Alice" should not be able to upload file "filesForUpload/lorem.txt" to "PARENT/textfile0.txt"
+    Examples:
+      | dav-path-version | lock-scope |
+      | old              | shared     |
+      | old              | exclusive  |
+      | new              | shared     |
+      | new              | exclusive  |
+      | spaces           | shared     |
+      | spaces           | exclusive  |
+
+
+  Scenario Outline: try to lock a file inside a folder shared by a link as anonymous user with read permission
+    Given using <dav-path-version> DAV path
+    And user "Alice" has created folder "PARENT"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "PARENT/textfile0.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | PARENT   |
+      | permissions | read     |
+      | password    | %public% |
+    When the public tries to lock "textfile0.txt" in the last public link shared folder using the new public WebDAV API setting the following properties
+      | lockscope | <lock-scope> |
+    Then the HTTP status code should be "403"
+    And 0 locks should be reported for file "PARENT/textfile0.txt" of user "Alice" by the WebDAV API
+    And user "Alice" should be able to upload file "filesForUpload/lorem.txt" to "PARENT/textfile0.txt"
+    Examples:
+      | dav-path-version | lock-scope |
+      | old              | shared     |
+      | old              | exclusive  |
+      | new              | shared     |
+      | new              | exclusive  |
+      | spaces           | shared     |
+      | spaces           | exclusive  |
```

Comment:
can this scenario be combined into 1?

Reply:
maybe I'll check

Reply:
@SagarGi No the scenario cannot be combined into one. One scenario is the case where the anon user has only `read ` permission so they should not be able to lock the file and another is where they have `edit` permission where they should be able to lock the file